src/hotspot/share/gc/g1/g1RedirtyCardsQueue.hpp

```diff
@@ -56,7 +56,7 @@ class G1RedirtyCardsLocalQueueSet : private PtrQueueSet {
   void enqueue(void* value);
 
   // Transfer all completed buffers to the shared qset.
-  void flush();
+  void flush(BufferNodeList* buffer_log);
```

Comment:
Please update the comment that the given pointer also gets returned the `BufferNodeList` describing the completed buffers as a slice of the shared qset.
Also not completely sure it's common for `flush` to return that and kind of add this here so the naming is not very good imo, but an improved comment should suffice.